demos/mcp_server_memory/memory_server.py

```
@@ -0,0 +1,316 @@
+                description="Primary topic/category for this content",
+                examples=["quantum_computing", "machine_learning", "history"]
+            )
+        ],
+        tags: Annotated[
+            List[str],
+            Field(
+                description="Optional tags for better retrieval",
+                default=[],
+                examples=[["physics", "computing", "technology"], ["ai", "neural_networks"]]
+            )
+        ] = []
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)